pywbem/_server.py

```diff
@@ -1122,3 +1320,65 @@ def _determine_profiles(self):
         mp_insts = self._conn.EnumerateInstances("CIM_RegisteredProfile",
                                                  namespace=self.interop_ns)
         self._profiles = mp_insts
+
+    def _count_associators(self):
+        """
+        Create count of associators for CIM_ReferencedProfile using the
+        antecedent and dependent reference properties as ResultRole for each
+        profile defined in server.profiles and return a dictionary of the
+        count. This code does a shortcut in executing EnumerateInstances to get
+        CIM_ReferencedProfile and processing the association locally.
+
+        Returns
+          Dictionary where the keys are the profile names and the value for
+          each key is a dictionary with two keys ('dep' and 'ant')
+          The value for each key is the count of associator paths when the key
+          value is used as the ResultRole
+
+        """
+        try:
+            ref_insts = self.conn.EnumerateInstances(
+                "CIM_ReferencedProfile", namespace=self.interop_ns)
+            # Remove host from responses since the host causes confusion
+            # with the enum of registered profile. Enums do not return host but
+            # the associator properties contain host
+            for ref_inst in ref_insts:
+                for prop in ref_inst.values():
+                    prop.host = None
+
+        except Error as er:
+            print('CIM_ReferencedProfile failed for server=%s exception=%s'
+                  % (self, er))
```

Comment:
Use new style formatting

Reply:
DONE

Reply:
We agreed to do the exception and put more context into the error here so that the user has overall operation context in the error.

Reply:
I want to implement a ServerError class for that.